### ISSUE TITLE
Support for urls with port numbers in them

### ISSUE THREAD

**Issue by jpann** [smorks/keepassnatmsg#88]
I would like support for matching urls that contain port numbers in them. I was previously using keepasshttp and it correctly matched urls that had ports. 

For example, if I am accessing LAN site that is at http://192.168.1.10:44306 and I have other entries in KeePass for sites at the following addresses: http://192.168.1.10:9779,  http://192.168.1.10:8443, keepassnatmsg lists logins for every site that that has http://192.168.1.10, regardless of the port specified and I'm only allowed to grant access to all of the entries. 

If I grant access, the `KeePassNatMsg Settings` plugin data for every entry listed entries is updated to include `{ "Allow" : ["192.168.1.10:44306"], "Deny" :[""] }` in addition to whichever other `Allowed` addresses where already there.

I've tried this with or without various settings like 'Return only best matching entries', 'match URL schemes' and 'search string fields for URL' and explicitly adding the URL and port there.

This also happens if I use local hostnames, e.g. http://mycomputer:44306, http://mycomputer:8443.

It does seem like once the URL is added to all of the listed entries, the next time I go back to that URL, it only displays the entry for that site and port.

KeePass v2.49
keepassnatmsg v2.0.14
Chrome and FireFox with keepassxc-browser

**Comment by smorks** [comment]
i thought this worked at some point too! i will investigate. thanks for the report!

**Comment by jpann** [comment]
FWIW, while I see this with IP addresses, I am mainly using local domain names (e.g. http://mycomputer.mydomain:8090, http://hostname.domain:9000/, etc)

**Comment by smorks** [comment]
i changed this to an enhancement, because after testing it, it appears that when initially prompting you, it shows all entries that match the domain/ip, regardless of the port. but once you allow & save, then it only shows the entries that match, is that the behavior you are seeing too?

i see you've attempted to fix it yourself, were you successful?

**Comment by jpann** [comment]
> i changed this to an enhancement, because after testing it, it appears that when initially prompting you, it shows all entries that match the domain/ip, regardless of the port. but once you allow & save, then it only shows the entries that match, is that the behavior you are seeing too?

Yes, but the matching domain/ip gets added to the KeePassNatMsg Settings  as an additional "Allow" property for every entry that has that domain/ip. If I have a second entry with the same domain/ip but a different port, it lists all of the entries with that matching domain/ip, like you mentioned, and then after I allow and save, the next time I go back to that site, the matched entry is for some other completely different entry with that domain/ip but a different port, but this time it is only a single entry and not a list.

> i see you've attempted to fix it yourself, were you successful?

So far it has been working OK, I have issues where I have to sometimes keep clicking on the "Fill Credentials" button before anything happens or sometimes the browser extension says it can't talk to the plugin, but I'm not sure if that is because of my change or just some weird environmental issue.

**Comment by smorks** [comment]
> So far it has been working OK, I have issues where I have to sometimes keep clicking on the "Fill Credentials" button before anything happens or sometimes the browser extension says it can't talk to the plugin, but I'm not sure if that is because of my change or just some weird environmental issue.

did you want to submit it as a pull request? I can do some testing on it too.

**Comment by diaodeng** [comment]
能麻烦处理下按端口匹配的问题吗，有很多域名相同但是端口不同的网站对应的账密，像下面这种输如一个地址会把下面两个都匹配出来

**Comment by Bond246** [comment]
I can confirm the same problem.

I have a lot of entrys for my local network with different port numbers. Normally i can allow or deny the request for a specific website and it will also write the port-number inside the deny-array but it doesn't work. The website seems to ignore the deny array of an entry.
